src/CHIPDriver.hh

```diff
@@ -118,4 +120,171 @@ extern std::atomic_ulong CHIPNumRegisteredFatBinaries;
  */
 extern thread_local hipError_t CHIPTlsLastError;
 
+class DeviceType {
+public:
+  enum Type { GPU, CPU, ACCEL, FPGA, DEFAULT };
+
+private:
+  Type Type_;
+
+public:
+  DeviceType(Type TypeIn) : Type_(TypeIn) {}
+
+  std::string str() const {
+    switch (Type_) {
+    case GPU:
+      return "gpu";
+    case CPU:
+      return "cpu";
+    case ACCEL:
+      return "accel";
+    case FPGA:
+      return "fpga";
+    case DEFAULT:
+      return "default";
+    default:
+      return "unknown";
+    }
+  }
+
+  Type getType() const { return Type_; }
+};
+
+class BackendType {
+public:
+  enum Type { OPENCL, LEVEL0, DEFAULT };
+
+private:
+  Type Type_;
+
+public:
+  BackendType(Type TypeIn) : Type_(TypeIn) {}
+
+  const char *str() const {
+    switch (Type_) {
+    case OPENCL:
+      return "opencl";
+    case LEVEL0:
+      return "level0";
+    case DEFAULT:
+      return "default";
+    default:
+      return "unknown";
+    }
+  }
+
+  Type getType() const { return Type_; }
+};
+
+class EnvVars {
+public:
+  int PlatformIdx = 0;
+  DeviceType Device = DeviceType::GPU;
+  int DeviceIdx = 0;
+  BackendType Backend = BackendType::DEFAULT;
+  bool DumpSpirv = false;
+  std::string JitFlags = CHIP_DEFAULT_JIT_FLAGS;
+  bool L0ImmCmdLists = true;
+  int L0CollectEventsTimeout = 0;
+
+  EnvVars() : Device(DeviceType::GPU), Backend(BackendType::DEFAULT) {
+    parseEnvironmentVariables();
+    logDebugSettings();
+  }
+
+private:
+  void parseEnvironmentVariables() {
+    // Parse all the environment variables and set the class members
+    if (!readEnvVar("CHIP_PLATFORM").empty())
+      PlatformIdx = parseInt("CHIP_PLATFORM");
+    Device = parseDeviceType("CHIP_DEVICE_TYPE");
+    if (!readEnvVar("CHIP_DEVICE").empty())
+      DeviceIdx = parseInt("CHIP_DEVICE");
+    Backend = parseBackendType("CHIP_BE");
+    if (!readEnvVar("CHIP_DUMP_SPIRV").empty())
+      DumpSpirv = parseBoolean("CHIP_DUMP_SPIRV");
+    JitFlags = parseJitFlags("CHIP_JIT_FLAGS_OVERRIDE");
+    if (!readEnvVar("CHIP_L0_IMM_CMD_LISTS").empty())
+      L0ImmCmdLists = parseBoolean("CHIP_L0_IMM_CMD_LISTS");
+    if (!readEnvVar("CHIP_L0_COLLECT_EVENTS_TIMEOUT").empty())
+      L0CollectEventsTimeout = parseInt("CHIP_L0_COLLECT_EVENTS_TIMEOUT");
+  }
+
+  std::string parseJitFlags(const std::string &StrIn) {
```

Comment:
Could we avoid returning a copy? E.g. by returning a const reference or std::string_view?